### PR TITLE
Do not store empty Bundles

### PR DESCRIPTION
**Pull request by Swatinem**
This avoids storing empty bundles as `ArtifactBundle`s, and also avoids keeping an orphaned `File` around when it is being extracted and stored as individual `ReleaseFile`s.

### CODE REVIEW

File: src/sentry/tasks/assemble.py [hunk 1]
```diff
@@ -485,10 +492,11 @@ def _validate_bundle(self):
             "tasks.assemble.artifact_bundle.artifact_count", amount=self.archive.artifact_count
         )
 
-    def close(self):
```
Comment (iambriccardo) [inline src/sentry/tasks/assemble.py:488]: Why did you remove this?

**Review by loewenheim — APPROVED** [review]

File: src/sentry/tasks/assemble.py (hunk 1 above)
Comment (iambriccardo) [reply src/sentry/tasks/assemble.py:488]: Oh now i see, looks good!

**Review by iambriccardo — APPROVED** [review]
